### DISCUSSION TITLE
Create the (possibility of) data model compatibility between Directus 8 and Next

### DISCUSSION THREAD

**Discussion by flyaruu** [directus/directus#3093] [Feature Requests]
While troubleshooting performance issues in Directus 8 I thought I take Directus Next for a spin. Running Directus Next upgrades the data model though, so Directus 8 stopped working.

If both could work off the same data model it would be way easier to move between versions, resulting in better alpha / beta feedback.

please disregard if it was already mentioned somewhere...

**Comment by benhaynes** [comment]
There are two sides to this, I will describe them both below:

## Your Project Data
Directus is unique in that it stores your data as pure, unaltered SQL. That means you always have the freedom to view, import and export your database content and data whenever you'd like. Therefore there is no migration required for ingesting legacy content into Directus, moving between versions of Directus, or even moving away from Directus. Each of those processes are as easy as opening your database in a DB Admin Tool and performing the import/export.

## Directus System Config
System data is stored decoupled and isolated from your project data, so there is no commingling or interference. However, unlike your custom data schema, our system data model is proprietary — that is to say, the base tables and fields are locked (yet extensible). 

We are a small open source team (at least until our monetized Cloud grows or we take on funding), and therefore have finite resources to commit to this free and open-source project. So we've chosen to prioritize having our most current version as stable and feature-rich as possible over writing complex up/down migration scripts between _major_ versions (we do always create them for patches and minors).

## TL;DR
You can easily import legacy SQL data into Directus, or migrate between major Directus versions... but you'll need to reconfigure your settings when switching between major versions of Directus. Alternatively, you can build a tool that automates this process and contribute it to the community... after all, this is open-source. 😉

**Comment by JLueke** [comment]
Is there any documentation on what changes where made to the database from 8 to 9?

Even though it is a major upgrade, I don't think many people will stick with Directus for long if they have to reconfigure the fields for 20+ collections after an upgrade. Even less if there is no documentation to be found. I currently compare the dumps, in hope that I only have to adapt the sql dump.

**Reply by benhaynes** [reply]
The Docs are coming, but we're focused on getting the platform stable first. I understand the frustration of not having a seamless upgrade process, but we've had the same architecture for a decade, and it was time to give it a major overhaul. That couldn't be done without making significant changes to the system schema. We decided to "rip off the bandaid", but that means there's no real way to migrate system data between versions. This is all a necessary part of keeping the platform modern and maintainable.

That said, your use content (arguably more important) migrates over 1:1 without issue. For the system stuff, we should have the migration article and v9 docs up soon (next week or so).

**Reply by JLueke** [reply]
> That said, your use content (arguably more important) migrates over 1:1 without issue.

I can't confirm this. Since the ids for user and file changed fom int(10) to char(36), all those references in my content tables need to be replaced.

**Reply by benhaynes** [reply]
True! Those two Collections are a mixture of system and user content. That said, an INT may also work in the UUID datatype, so perhaps that's not an issue (other than the annoyance that some PKs will be numbers, and others will be UUIDs. We're not ruling out a migration assistant, we just can't get that built now since we need to get v9 fully stable first.

**Reply by JLueke** [reply]
I got permission errors when I tried for thumbnails and file downloads, when I used the old ids. Only worked after I replaced the id with the one from the filename_disk column (without extension).

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
